### PR TITLE
Fix wrapText handling for long unbreakable words

### PR DESCRIPTION
**Pull request by kadireren7**
## Summary

Fix invalid pointer arithmetic in `wrapText()` when no whitespace is found within the wrapping window.

Previously, `lastspace` could be `NULL`, but it was still used to calculate the next text position. This caused undefined behavior and truncated the remaining debugger help text.

## Changes

- Advance by `max_len` when no whitespace is found
- Add a debugger command fixture with a long unbreakable description
- Add a regression test for the debugger help output

## Testing

- Scripting and functions test suites: 766 passed
- Scripting engine tests: 39 passed
- `git diff --check`: clean
- UBSan reproduced the invalid pointer arithmetic in an isolated test

Fixes debugger help text truncation for long descriptions without whitespace.

### CODE REVIEW

File: src/scripting_engine.c [hunk 1]
```diff
@@ -829,9 +829,13 @@ static sds *wrapText(const char *text, size_t max_len, size_t *count) {
             char *lastspace = strrchr(line, ' ');
             if (lastspace != NULL) {
                 *lastspace = 0;
+                p += (lastspace - line) + 1;
+            } else {
+                /* If no whitespace is found in the current window, advance by
+                 * max_len. Using lastspace when it is NULL performs invalid
+                 * pointer arithmetic and previously truncated the remaining text. */
+                p += max_len;
```
[the remark this answers is not in the document]
Comment (kadireren7) [reply src/scripting_engine.c:835]: Confirmed and fixed. I added whitespace skipping after the hard split and extended the regression coverage with a boundary case where the separator appears exactly at max_len. The updated tests fail without the fix and pass with it. All 39 scriptingengine tests and 766 scripting/functions tests pass.

File: src/scripting_engine.c [hunk 2]
```diff
@@ -829,9 +829,12 @@ static sds *wrapText(const char *text, size_t max_len, size_t *count) {
             char *lastspace = strrchr(line, ' ');
             if (lastspace != NULL) {
                 *lastspace = 0;
+                p += (lastspace - line) + 1;
+            } else {
+                /* No whitespace found; hard-split at max_len. */
+                p += max_len;
+                while (*p == ' ') p++;
```
[the remark this answers is not in the document]
Comment (kadireren7) [reply src/scripting_engine.c]: Confirmed and fixed. I moved the whitespace skip after the split branches so it applies to both whitespace-based and hard splits, and added a regression case for adjacent spaces at the wrap boundary. The regression fails without the fix and passes with it. All 39 scriptingengine tests and 766 scripting/functions tests pass.
